Cutelyst/Plugins/Utils/Validator/validator.cpp

```diff
@@ -57,36 +57,49 @@ void Validator::clear()
     }
 }
 
-bool Validator::validate(Context *c, ValidatorFlags flags) const
+Cutelyst::ValidatorResult Validator::validate(Context *c, ValidatorFlags flags) const
 {
     if (!c) {
         qCWarning(C_VALIDATOR) << "No valid Context set, aborting validation.";
-        return false;
+        return ValidatorResult();
     }
 
-    const bool valid = validate(c->request()->parameters(), flags);
+    const ParamsMultiMap params = c->request()->parameters();
+    const ValidatorResult result = validate(params, flags);
```

Comment:
The Return Value Optimizaion on some compilers require that the Return value is at the top and Returning ValidatorResult() would also break that optimization, so try changing to a code like
>{
  ValidatorResult result;
  if(){
    return result
}`
result = validate();
}

I'm not completely sure when it works or not, but I try to write code as docs say so :P